vsphere/datadog_checks/vsphere/vsphere.py

```diff
@@ -555,9 +556,32 @@ def collect_events(self):
             # OR something bad happened (which might happen again indefinitely).
             self.latest_event_query = collect_start_time
 
+    def get_server_current_time(self):
+        # type: (Any) -> dt.datetime
+        """
+        Return server current datetime.
+        If the retrieval of server current datetime fails, use local current datetime.
+
+        The server current datetime is cached for a check run.
+        """
+        if self._server_current_time is None:
```

Comment:
I think the caching part of the server time makes it a bit too complex and error-prone if we forget to reset the server current time to None between check runs.

Instead could we remove that method and that property altogether and call `self.api.get_current_time()` once at the beginning of the `make_query_specs` method?

Reply:
Yes, agree, it might not be necessary to cache.

Reply:
@FlorianVeaux addressed